pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -180,9 +181,19 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 				return reconcile.Result{}, err
 			}
 		}
-
+		// look-up the `basic` NSTemplateTier to get the NS templates
+		var nstemplateTier toolchainv1alpha1.NSTemplateTier
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: request.Namespace, // assume that NSTemplateTier were created in the same NS as Usersignups
+			Name:      "basic",
+		}, &nstemplateTier)
+		if err != nil {
+			reqLogger.Error(err, "No 'basic' NSTemplateTier found.")
+			// let's requeue until the NSTemplateTier resource is available
+			return reconcile.Result{Requeue: true}, r.updateStatus(reqLogger, instance, r.setStatusNoTemplateTierAvailable)
```

Comment:
Use `wrapErrorWithStatusUpdate()` instead. So we have the original error in the status too. See  https://github.com/codeready-toolchain/host-operator/blob/b2a7ccf79f1baff97da785689f2480bbfff65621/pkg/controller/usersignup/usersignup_controller.go#L147 for example.

Reply:
done in f29652e